source/auth/auth.rst

```diff
@@ -683,6 +683,176 @@ mechanism
+d    X-AMZ-Date           string     Current date in UTC -see `AWS Signature Version 4 <https://docs.aws.amazon.com/general/latest/gr/signature-version-4.html?shortFooter=true>`_
+t    X-AMZ-Security-Token string     Optional AWS security token
+==== ==================== ========== ============================================================================================================================================== 
+
+Drivers MUST NOT advertise support for channel binding, as the server does
+not support it and legacy servers may fail authentication if drivers advertise
+support. The client-first-message MUST set ``p`` to the integer representation 
+of the ASCII character ``n``, i.e., ``110``.
+
+Conversation
+````````````
+As an example, given an ``r`` value of "enJwWTtNSkR+WztFZCE3d1NWSiMpfU54YCgmPU5lY1Q=", a MONGODB-IAM conversation would appear as
+follows:
+
+| C: ``r=enJwWTtNSkR+WztFZCE3d1NWSiMpfU54YCgmPU5lY1Q=,p=110``
+| S: ``s=enJwWTtNSkR+WztFZCE3d1NWSiMpfU54YCgmPU5lY1RHbnN1IWy6vp7GvmtRmcGWYEtjedGEI0ZXi13r7y4V+A==,h="sts.amazonaws.com"``
+
+.. note::
+	Drivers MUST validate that the server nonce is exactly 64 bytes and the first 32 bytes are the same as the client nonce, i.e., ``r``. Drivers must 
+        also validate that the host is greater than 0 and less than or equal to 255 bytes per `RFC 1035 <https://tools.ietf.org/html/rfc1035>`_. Drivers MUST reject
+        DNS names with empty labels, e.g., "abc..def" and error on any additional fields.
+
+| C: ``a="AWS4-HMAC-SHA256 Credential=AKIAICGVLKOKZVY3X3DA/20191017/us-east-1/sts/aws4_request, SignedHeaders=content-length;content-type;host;x-amz-date;x-mongodb-gs2-cb-flag;x-mongodb-server-nonce, Signature=6f67c4dfbc05da04ea8d571bc6f7d11833f9d1909d0068be302c5379e1867d15",d="20191017T173547Z"``
+
+Signature Version 4 Signing Process
+        In response to the Server First message, the driver MUST construct a call to `sts::GetCallerIdentity <https://docs.aws.amazon.com/STS/latest/APIReference/API_GetCallerIdentity.html>`_ with
+        its credentials including the base 64 representation of the server nonce, .i.e., ``s``, as a signed header named ``X-MongoDB-Server-Nonce``. It must 
+        also include ``X-MongoDB-GS2-CB-Flag`` as a signed header. The value of ``X-MongoDB-GS2-CB-Flag`` MUST be the single ASCII character ``n``. These headers must be included in the list
+        of ``SignedHeaders`` in the authorization header, i.e., ``a``, of the AWS Signature V4. There is one optional field, ``X-Amz-Security-Token``, which are temporary credentials used
+        to generate the signature. The values that drivers should use to generate the authorization signature are as follows:
+
+======================== ======================================================================================================
+Name                     Value       
```

Comment:
I'm guessing the order in which these appear in the `SignedHeaders` matters since we only pass along values and not the field names? In that case, can we state somewhere what the expected order for these values to appear in the signed header is? At present, there are some entries in the table that don't show up in the example on L731 so its a bit hard to glean where to put those values.

Reply:
Added clarity on how to construct the authorization header.

Reply:
@prashantmital please comment back or resolve.